helix-core/src/selection.rs

```diff
@@ -54,6 +54,9 @@ pub struct Range {
     /// The head of the range, moved when extending.
     pub head: usize,
     pub horiz: Option<u32>,
+    /// True if range is intended to contain whole lines (e.g. after `line_extend_below`).
+    /// Useful to prevent first `line_extend_below` on an empty line selecting the next line.
+    pub line_mode: bool,
```

Comment:
This should be a field on the Selection, not Range?

Reply:
I have no idea. :) . Seem to work OK for me, AFAICT.